org.eclipse.ice.dev/org.eclipse.ice.dev.annotations/src/main/java/org/eclipse/ice/dev/annotations/processors/Field.java

```diff
@@ -134,6 +134,50 @@ public String getNameForMethod() {
 		return StringUtils.capitalize(this.name);
 	}
 
+	/**
+	 * Return the appropriate getter method name for this field.
+	 * @return getter method name
+	 */
+	@JsonIgnore
+	public String getGetterName() {
+		String prefix = null;
+		if (type != null && type.equals("boolean")) {
+			prefix = "is";
+		} else {
+			prefix = "get";
+		}
+		return prefix + getNameForMethod();
+	}
+
+	/**
+	 * Return whether this field has a getter, directly or via one of its aliases.
+	 * @return true if getter present, false otherwise
+	 */
+	@JsonIgnore
+	public boolean hasGetter() {
```

Comment:
What is the reason for having this function only call another function? Are both used in different ways? I guess my question is, could they be combined into a single function, and if not why?

Reply:
Good question. This method is intended for use from the velocity template where holding on to values is possible but not necessarily very clean. Having the separate boolean returning method reduces the amount of logic required in the template and, if logic changes in the future, that change would be transparent to the template. As an example of where this is used in the template, here is the copy method template:
```
	/**
	 * Copy the contents of another element into this element.
	 * @param element the element to copy
	 */
	@Override
	public void copy(@NonNull $interface element) {
		#foreach($field in $fields)
			#if($field.hasGetter())
		this.${field.Name} = element.${field.AnyGetter}();
			#end
		#end
	}
```

Without the separate `hasGetter()`:
```
	/**
	 * Copy the contents of another element into this element.
	 * @param element the element to copy
	 */
	@Override
	public void copy(@NonNull $interface element) {
		#foreach($field in $fields)
			#set($getter = ${field.AnyGetter})
			#if($getter)
		this.${field.Name} = element.$getter();
			#end
		#end
	}
```
Again, possible but, in my opinion at least (and I'm not super set on this), having the boolean `hasGetter()` seems better.